cmd/zoekt-sourcegraph-indexserver/cleanup.go

```diff
@@ -98,19 +109,25 @@ func cleanup(indexDir string, repos []uint32, now time.Time, shardMerging bool)
 	}
 
 	// index: Move missing repos from trash into index
+	// index: Restore deleted or tombstoned repos.
 	for _, repo := range repos {
 		// Delete from index so that index will only contain shards to be
 		// trashed.
 		delete(index, repo)
 
-		shards, ok := trash[repo]
-		if !ok {
-			continue
+		if shards, ok := trash[repo]; ok {
+			log.Printf("restoring shards from trash for %v", repo)
+			moveAll(indexDir, shards)
+			shardsLog(indexDir, "restore", shards)
 		}
 
-		log.Printf("restoring shards from trash for %v", repo)
-		moveAll(indexDir, shards)
-		shardsLog(indexDir, "restore", shards)
+		if shardMerging {
```

Comment:
meta: do we still need this variable? ie, can't we just rely on the fact a shard is a compound shard rather than gating?

Reply:
yes, you are right. If shard merging is not enabled, tombstones will be a nil map and we skip the block anyway.